plugins/woocommerce/src/Internal/Admin/EmailPreview/EmailPreview.php

```diff
@@ -101,16 +141,22 @@ private function render_legacy_preview_email() {
 	private function render_preview_email() {
 		$this->set_up_filters();
 
-		$email = $this->get_email();
+		$emails = WC()->mailer()->get_emails();
+		$email  = $emails[ $this->email_type ];
 
 		$order = $this->get_dummy_order();
-		$email->set_object( $order );
+		/**
+		 * An object used in email preview. Defaults to a dummy WC_Order.
+		 *
+		 * @since 9.6.0
+		 */
+		$email->set_object( apply_filters( 'woocommerce_email_preview_object', $order ) );
```

Comment:
Because we already have the `woocommerce_email_preview_dummy_order` filter available, I don't think this second filter on the dummy order is needed.

From a WooCommerce Subscriptions perspective, we need to be able to filter the `$email` object so we can set custom variables before `email->get_content_html();` is called.

For example, see our [`WCS_Email_New_Switch_Order->get_content_html()`](https://github.com/Automattic/woocommerce-subscriptions-core/blob/1692c41cf43e2c3c5ef0816a48196ffb241b9e72/includes/emails/class-wcs-email-new-switch-order.php#L121) function which requires `$this->object` and `$this->subscriptions` to be set.

With these new filters, I don't think we have a way to manually set `$this->subscriptions`.

Thoughts on introducing a method to filter the `$email` object, i.e. something like:

```
$email = apply_filters( 'woocommerce_prepare_email_for_preview', $email );
```

I'm happy to make changes to WooCommerce Subscriptions to better support this as well if we don't want to introduce that filter.

Reply:
Great feedback, @mattallan! You already mentioned the need to modify `$email` before, but it kind of slipped my mind. Updated in 0f28cf7.